### DISCUSSION TITLE
Add event to toggle Todo List

### DISCUSSION THREAD

**Discussion by JulesCBB** [ueberdosis/tiptap#878] [General Questions] [answered]
Hey guys,

I've started working with the Tip Tap editor recently and I've been wanting to modify the TodoList component so that when a user enters {} (left curly bracket, right curly bracket) on the text editor, it toggles the todo list. However, I'm not really sure how to do this. Does anyone have any suggestions?

Thanks in advance!

**Comment by hanspagel** [comment, marked as answer]
Hi JulesCBB! You probably want to change the input rule for the TodoList extension. Have a look at the source code:

https://github.com/ueberdosis/tiptap/blob/main/packages/tiptap-extensions/src/nodes/TodoList.js

You can extend that class, overwrite the input rules (replace square brackets with curly brackets in the regex) and register your custom extension with tiptap.

Hope that helps!

**Reply by JulesCBB** [reply]
Thanks a lot, it worked! For some reason I was a bit confused whether to modify that method or keys().

**Reply by hanspagel** [reply]
Thanks! I can understand that. We’ll make it more clear in the upcoming documentation.
